File: android/app/src/main/java/com/tonhub/wallet/modules/wallet/ProvisionRequest.java
package com.tonhub.wallet.modules.wallet;

import java.util.concurrent.CompletableFuture;

public class ProvisionRequest {
    public boolean isTestnet;
    public String token;
    public String cardId;
    public CompletableFuture<Boolean> completableFuture;
    public String displayName;
    public String lastDigits;

    public ProvisionRequest(boolean isTestnet, String token, String cardId, String displayName, String lastDigits, CompletableFuture<Boolean> completableFuture) {
        this.isTestnet = isTestnet;
        this.token = token;
        this.displayName = displayName;
        this.lastDigits = lastDigits;
        this.cardId = cardId;
        this.completableFuture = completableFuture;
    }
}

File: android/app/src/main/java/com/tonhub/wallet/modules/wallet/ProvisionRequest.kt
package com.tonhub.wallet.modules.wallet

import java.util.concurrent.CompletableFuture

data class ProvisionRequest(
    val isTestnet: Boolean,
    val token: String,
    val cardId: String,
    val displayName: String,
    val lastDigits: String,
    val completableFuture: CompletableFuture<Boolean>
)


